### ISSUE TITLE
Colormap: interpolate from collection of colors

### ISSUE THREAD

**Issue by swharden** [ScottPlot/ScottPlot#4247]
There should be a way to create a colormap starting from a collection of colors. I don't recall whether this is already possible. 

Related, `Color.InterpolateRgbArray`:
https://github.com/ScottPlot/ScottPlot/blob/743c9427e63692d31920cc424a0e62b12d6eec4a/src/ScottPlot5/ScottPlot5/Primitives/Color.cs#L387-L396

Note that similar functionality exists in Python
* https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.ListedColormap.html#matplotlib.colors.ListedColormap
* https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.LinearSegmentedColormap.html#matplotlib.colors.LinearSegmentedColormap

**Comment by swharden** [comment]
`Colormap.Custom` is close, but doesn't interpolate between adjacent colors

https://github.com/ScottPlot/ScottPlot/blob/743c9427e63692d31920cc424a0e62b12d6eec4a/src/ScottPlot5/ScottPlot5/Colormaps/Custom.cs#L3-L20

**Comment by swharden** [comment]
Here's some code to get this started

```cs
Color[] colors = { Colors.Orange, Colors.Gray, Colors.Blue };
IColormap colormap = new LinearSegmented(colors);

ScottPlot.Plot plot = new();
double[] ys = ScottPlot.Generate.Sin(51);
for (int i = 0; i < 10; i++)
{
    var sig = plot.Add.Signal(ys);
    sig.AlwaysUseLowDensityMode = true;
    sig.Color = colormap.GetColor(i, 10);
    sig.Data.XOffset = i * 2;
    sig.LineWidth = 3;
}
plot.SavePng("test.png", 400, 300).LaunchInBrowser();
```

![image](https://github.com/user-attachments/assets/515e2886-0b83-48fc-96ff-55193d39cdcb)

```cs
public class LinearSegmented : ScottPlot.Colormaps.ColormapBase, IColormap
{
    public override string Name => "Linear Segmented";
    private readonly Color[] Colors;

    public LinearSegmented(Color[] colors)
    {
        ArgumentNullException.ThrowIfNull(colors);

        if (colors.Length < 2)
            throw new ArgumentException($"{nameof(colors)} must contain at least 2 colors");

        Colors = colors;
    }

    public override Color GetColor(double position)
    {
        if (position == 0)
            return Colors[0];

        if (position == 1)
            return Colors[^1];

        int rangeCount = Colors.Length - 1;
        double rangeSize = 1.0 / rangeCount;

        int firstColorIndex = (int)(position / rangeSize);
        double positionInRange = (position - (rangeSize * firstColorIndex)) / rangeSize;

        Color color1 = Colors[firstColorIndex];
        Color color2 = Colors[firstColorIndex + 1];

        return color1.MixedWith(color2, positionInRange);
    }
}
```

**Comment by swharden** [comment]
While we're in there, I want to add a new colormap to hold the following values. I made this many years ago as a modified version of Turbo with less intense yellows and greens so thin traces are easier to evaluate on a light background

```cs
Color[] colors = "#466be3 #29bbec #30f199 #edd03a #fb8023 #d23104"
    .Split(" ")
    .Select(x => new Color(x))
    .ToArray();
```

![image](https://github.com/user-attachments/assets/2d5cde28-6f49-4f7c-bf03-338d769a5e05)

**Comment by swharden** [comment]
resolved in #4324
